Firmware/LowLevel/src/datatypes.h

```diff
@@ -135,22 +137,62 @@ struct ll_ui_event {
+};
+
+#pragma pack(push, 1)
+struct HallConfig {
+    HallMode mode : 4;
+    HallLevel level : 1;
+} __attribute__((packed));
+#pragma pack(pop)
+
+#define MAX_HALL_INPUTS 10  // How much Hall-inputs we support. 4 * OM + 6 * Stock-CoverUI + 0 spare (because not yet required to make it fixed)
+
+// LL/HL config packet, bi-directional, flexible-length, with defaults for YF-C500.
 #pragma pack(push, 1)
 struct ll_high_level_config {
-    uint8_t type;
-    uint8_t comms_version = LL_HIGH_LEVEL_CONFIG_MAX_COMMS_VERSION;  // Increasing comms packet-version number for packet compatibility (n > 0)
-    uint8_t config_bitmask = 0;                                      // See LL_HIGH_LEVEL_CONFIG_BIT_*
-    int8_t volume;                                                   // Volume (0-100%) feedback (if directly changed via CoverUI)
-    iso639_1 language;                                               // ISO 639-1 (2-char) language code (en, de, ...)
-    uint16_t spare1 = 0;                                             // Spare for future use
-    uint16_t spare2 = 0;                                             // Spare for future use
-    uint16_t crc;
+    // ATTENTION: This is a flexible length struct. It is allowed to grow independently to HL without loosing compatibility,
+    //    but never change or restructure already published member, except you really know their consequences.
+
+    // uint8_t type; Just for illustration. Get set in wire buffer with type PACKET_ID_LL_HIGH_LEVEL_CONFIG_REQ or PACKET_ID_LL_HIGH_LEVEL_CONFIG_RSP
+
+    uint8_t config_bitmask = 0;                    // See LL_HIGH_LEVEL_CONFIG_BIT_*
+    uint16_t rain_threshold = 700;                 // If (stock CoverUI) rain value < rain_threshold then it rains. Expected to differ between C500, SA and SC types
+    float v_charge_cutoff = V_CHARGE_MAX;          // Protective max. charging voltage before charging get switched off
+    float i_charge_cutoff = I_CHARGE_MAX;          // Protective max. charging current before charging get switched off
+    float v_battery_cutoff = 29.0f;                // Protective max. battery voltage before charging get switched off
+    float v_battery_empty = BATT_EMPTY;            // Empty battery voltage used for % calc of capacity
+    float v_battery_full = BATT_FULL;              // Full battery voltage used for % calc of capacity
+    uint16_t lift_period = LIFT_EMERGENCY_MILLIS;  // Period (ms) for both wheels to be lifted in order to count as emergency (0 disable, 0xFFFF do not change). This is to filter uneven ground
+    uint16_t tilt_period = TILT_EMERGENCY_MILLIS;  // Period (ms) for a single wheel to be lifted in order to count as emergency (0 disable, 0xFFFF do not change). This is to filter uneven ground
+    iso639_1 language = {'e', 'n'};                // ISO 639-1 (2-char) language code (en, de, ...)
+    int8_t volume = 80;                            // Volume (0-100%) feedback (if directly changed i.e. via CoverUI or WebApp)
```

Comment:
Maybe use `uint8_t`?

Reply:
No, need int. It's an option :-)
Sound volume can be used as leading by ROS or leading by mower.
ROS send -1 if it doesn't modify the current setting